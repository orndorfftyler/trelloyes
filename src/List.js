import React from 'react';
import Card from './Card.js';

function List(props) {
    const neededCards = props.cardIds;
    
    const listCards = neededCards.map(object => 
        <Card
        key={object.id}
        id={object.id}
        title={object.title}
        content={object.content}
        />
        );
    return (
        <section className="List">
        <header className="List-header">
          <h2>{props.header}</h2>
        </header>
        <div key={props.id} id={props.id} className="List-cards">
            {listCards}
        </div>
        </section>
    );
}

export default List;