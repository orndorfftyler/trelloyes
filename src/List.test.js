import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

describe('List component', () => {

  it('renders without crashing', () => {
    let cardIds = [{id:"l",title:"Twelfth card",content:"lorem ipsum"},{id:"m",title:"Thirteenth card",content:"lorem ipsum"}];
    const div = document.createElement('div');
    ReactDOM.render(<List key='4' id='4' header='Fourth list' cardIds={cardIds} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it ('renders the UI as expected', () => {
    let cardIds = [{id:"l",title:"Twelfth card",content:"lorem ipsum"},{id:"m",title:"Thirteenth card",content:"lorem ipsum"}];
    const tree = renderer
      .create(<List key='4' id='4' header='Fourth list' cardIds={cardIds} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});
