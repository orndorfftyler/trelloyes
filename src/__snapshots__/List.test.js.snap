// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Fourth list
    </h2>
  </header>
  <div
    className="List-cards"
    id="4"
  >
    <div
      className="Card"
      id="l"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Twelfth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
      id="m"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Thirteenth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;
